atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/AbstractDbReadTable.java

```diff
@@ -210,10 +211,21 @@ protected AbstractDbReadTable(ConnectionSupplier conns,
     }
 
     @Override
-    public ClosableIterator<AgnosticLightResultRow> getRowsColumnRange(List<byte[]> rows, long ts, ColumnRangeSelection columnRangeSelection) {
-        FullQuery query = queryFactory.getRowsColumnRangeQuery(rows, ts, columnRangeSelection);
+    public ClosableIterator<AgnosticLightResultRow> getRowsColumnRangeCounts(List<byte[]> rows,
+                                                                             long ts,
+                                                                             ColumnRangeSelection columnRangeSelection) {
+        FullQuery query = queryFactory.getRowsColumnRangeCountsQuery(rows, ts, columnRangeSelection);
         AgnosticLightResultSet results = conns.get().selectLightResultSetUnregisteredQuery(query.getQuery(), query.getArgs());
-        results.setFetchSize(columnRangeSelection.getBatchHint() * rows.size());
+        results.setFetchSize(rows.size());
```

Comment:
We may want to put an upper bound on this fetch size so that for large rows we're not waiting on a ton of data over the wire.